src/compas/robots/resources/basic.py

```diff
@@ -169,6 +171,34 @@ def _get_local_path(self, url):
         return self.build_path(*path.split('/'))
 
 
+class LocalFolderMeshLoader(AbstractMeshLoader):
```

Comment:
I'm not really convinced about this additional loader. It does almost the same thing than the default one, and that can probably be the reason why you had to push the default one to the end of the list. That is likely to cause more confusion down the road because they are really really close in terms of what they do. What would be needed on the default one so that it can handle also this case you have?

Reply:
I agree on this one, because indeed it is very similar to the default loader. What I am trying to address is the scenario where the `shape.filename` is relative to the directory where the loaded `urdf` file lives. However, I haven't found a way to store it and then retrieve it later on as the `RobotModel` class doesn't store this information when parsed from `urdf`. Could it be that we store the `urdf`directory in `RobotModel.attr` dictionary, and query it afterwards? Although I am aware storing the `urdf` directory is not strictly necesary if the geometry can be processed with the other loaders.

Reply:
Actually, would passing some `kwargs` to `DefaultMeshLoader` do? Say, for example:

```suggestion
class DefaultMeshLoader(AbstractMeshLoader):
    def __init__(self, **kwargs):
        super(DefaultMeshLoader, self).__init__()
        self.attr = dict() or kwargs

    def can_load_mesh(self, url):
        url = self._get_local_path(url)
        scheme = urlparse(url).scheme
        # continues

    def _get_local_path(self, url):
        filepath = self.attr.get('filepath') 
        if filepath:
            return os.path.join(filepath, url)
        return url
```